### PR TITLE
Replace uninterruptible `Lock.lock` with `Lock.lockInterruptibly`

### PR DESCRIPTION
**Pull request by stIncMale**
JAVA-5140

### CODE REVIEW

**Review by jyemin — APPROVED** [review]
One optional change request.

LGTM

File: driver-core/src/main/com/mongodb/internal/connection/ConcurrentPool.java [hunk 1]
```diff
@@ -515,15 +515,7 @@ static String sizeToString(final int size) {
...
-    static void lockInterruptibly(final Lock lock) throws MongoInterruptedException {
-        try {
-            lock.lockInterruptibly();
-        } catch (InterruptedException e) {
-            throw interruptAndCreateMongoInterruptedException(null, e);
-        }
-    }
-
-    private static void lockInterruptiblyUnfair(final ReentrantLock lock) throws MongoInterruptedException {
+    static void lockInterruptiblyUnfair(final ReentrantLock lock) throws MongoInterruptedException {
```
[the remark this answers is not in the document]
Comment (stIncMale) [reply driver-core/src/main/com/mongodb/internal/connection/ConcurrentPool.java]: Done.

**Review by vbabanin — APPROVED** [review]
LGTM!

**Review by katcharov — APPROVED** [review]
LGTM

**Pull request merged**